corehq/apps/userreports/sql/adapter.py

```diff
@@ -211,8 +211,11 @@ def bulk_delete(self, docs, use_shard_col=True):
         table = self.get_table()
         doc_ids = [doc['_id'] for doc in docs]
         delete = table.delete(table.c.doc_id.in_(doc_ids))
-        with self.session_context() as session:
-            session.execute(delete)
+        try:
+            with self.session_context() as session:
+                session.execute(delete)
+        except ProgrammingError:
```

Comment:
Thanks for the revisit to the approach. Since ProgrammingError could come for different reasons, can we combine your original solution as
```
except ProgrammingError as e:
    if not self.table_exists:
        return
    else:
        raise e
```
```

Reply:
@mkangia Cool, let me make some updates

Reply:
https://github.com/dimagi/commcare-hq/pull/34595/commits/629697510cd007def6c92790f6fd6db69081e146